### PR TITLE
Fixviolations

### PR DESCRIPTION
**Pull request by rutgervandenberg**
Fix checkstyle/pmd/findbugs violations.
Some violations were false positives or had a good reason to exist. These were suppressed.
Checkstyle config was modified to allow for non-final methods. This allows for mocking and extending methods.

### CODE REVIEW

**Comment by albert-smit** [comment]
looks good

File: lifetiles-core/src/main/java/nl/tudelft/lifetiles/notification/model/ErrorNotification.java [hunk 1]
```diff
@@ -35,7 +35,7 @@ public ErrorNotification(final String message) {
      * {@inheritDoc}
      */
     @Override
-    public final Color getColor() {
+    public Color getColor() {
         return Color.ORANGERED;
```
Comment (jorenham) [inline lifetiles-core/src/main/java/nl/tudelft/lifetiles/notification/model/ErrorNotification.java:39]: Should be a constant

File: lifetiles-core/src/main/java/nl/tudelft/lifetiles/notification/model/ErrorNotification.java [hunk 2]
```diff
@@ -43,7 +43,7 @@ public final Color getColor() {
      * {@inheritDoc}
      */
     @Override
-    public final int getPriority() {
+    public int getPriority() {
```
Comment (jorenham) [inline lifetiles-core/src/main/java/nl/tudelft/lifetiles/notification/model/ErrorNotification.java:46]: Writing a getter for a constant is redundant, just make the constant public

File: lifetiles-core/src/main/java/nl/tudelft/lifetiles/notification/model/InfoNotification.java [hunk 3]
```diff
@@ -36,7 +36,7 @@ public InfoNotification(final String message) {
      * {@inheritDoc}
      */
     @Override
-    public final Color getColor() {
+    public Color getColor() {
         return Color.AQUA;
```
Comment (jorenham) [inline lifetiles-core/src/main/java/nl/tudelft/lifetiles/notification/model/InfoNotification.java:40]: Should be a constant

File: lifetiles-core/src/main/java/nl/tudelft/lifetiles/notification/model/InfoNotification.java [hunk 4]
```diff
@@ -44,7 +44,7 @@ public final Color getColor() {
      * {@inheritDoc}
      */
     @Override
-    public final int getPriority() {
+    public int getPriority() {
         return PRIORITY;
```
Comment (jorenham) [inline lifetiles-core/src/main/java/nl/tudelft/lifetiles/notification/model/InfoNotification.java:48]: Writing a getter for a constant is redundant, just make the constant public

File: lifetiles-core/src/main/java/nl/tudelft/lifetiles/notification/model/WarningNotification.java [hunk 5]
```diff
@@ -43,7 +43,7 @@ public final Color getColor() {
      * {@inheritDoc}
      */
     @Override
-    public final int getPriority() {
+    public int getPriority() {
```
Comment (jorenham) [inline lifetiles-core/src/main/java/nl/tudelft/lifetiles/notification/model/WarningNotification.java:46]: Writing a getter for a constant is redundant, just make the constant public

File: lifetiles-graph/src/main/java/nl/tudelft/lifetiles/graph/controller/GraphController.java [hunk 6]
```diff
@@ -218,15 +234,17 @@ private void initListeners() {
...
-            visibleSequences = (Set<Sequence>) args[0];
-            model.setVisible(visibleSequences);
-            diagram = new StackedMutationContainer(model.getBucketCache(),
-                    visibleSequences);
-            repaintNow = true;
-            repaint();
-        });
+            assert args[0] instanceof Set<?>;
+            // unfortunately java doesn't really let us typecheck generics :(
+                @SuppressWarnings("unchecked")
```
Comment (jorenham) [inline lifetiles-graph/src/main/java/nl/tudelft/lifetiles/graph/controller/GraphController.java:239]: Bad indentation

File: lifetiles-graph/src/main/java/nl/tudelft/lifetiles/graph/model/BucketCache.java [hunk 7]
```diff
@@ -71,7 +72,10 @@ private void cacheGraph() {
         Timer timer = Timer.getAndStart();
         buckets = new ArrayList<SortedSet<SequenceSegment>>();
         for (int index = 0; index < numberBuckets; index++) {
-            buckets.add(index, new TreeSet<SequenceSegment>());
+            // We do actually need to instantiate here.
+            @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
+            TreeSet<SequenceSegment> newSegments = new TreeSet<SequenceSegment>();
```
Comment (jorenham) [inline lifetiles-graph/src/main/java/nl/tudelft/lifetiles/graph/model/BucketCache.java]: Don't use `TreeSet` as type

File: lifetiles-graph/src/main/java/nl/tudelft/lifetiles/graph/model/DefaultGraphParser.java [hunk 8]
```diff
@@ -64,7 +64,10 @@ private SequenceSegment createSegment(final String descriptor,
         for (String sequencename : sources) {
             sequencename = sequencename.trim();
             if (!sequences.containsKey(sequencename)) {
-                sequences.put(sequencename, new DefaultSequence(sequencename));
+                // This is a parser. We do actually need to instantiate here.
+                @SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
+                DefaultSequence newSequence = new DefaultSequence(sequencename);
```
Comment (jorenham) [inline lifetiles-graph/src/main/java/nl/tudelft/lifetiles/graph/model/DefaultGraphParser.java]: Don't use `DefaultSequence` as type

File: lifetiles-sequence/src/main/java/nl/tudelft/lifetiles/sequence/controller/SequenceController.java [hunk 9]
```diff
@@ -102,30 +97,35 @@ public void initialize(final URL location, final ResourceBundle resources) {
...
-                        return;
-                    }
-                    assert (args[0] instanceof Map<?, ?>);
+        listen(Message.LOADED, (sender, subject, args) -> {
+            if (!"sequences".equals(subject)) {
+                return;
+            }
+            // eclipse and PMD disagree on whether parentheses are
+            // needed
+                assert args[0] instanceof Map<?, ?>;
```
Comment (jorenham) [inline lifetiles-sequence/src/main/java/nl/tudelft/lifetiles/sequence/controller/SequenceController.java:115]: bad indentation

File: lifetiles-tree/src/main/java/nl/tudelft/lifetiles/tree/controller/TreeController.java [hunk 10]
```diff
@@ -82,31 +78,37 @@ public final void initialize(final URL location,
...
         });
 
         // inform the sunburst of this controller so filters can be shouted
         view.setController(this);
         view.setBounds(wrapper.layoutBoundsProperty().get());
-        wrapper.layoutBoundsProperty().addListener((observableBounds, oldValue, newValue) -> {
-            view.setBounds(newValue);
-        });
+        wrapper.layoutBoundsProperty().addListener(
+                (observableBounds, oldValue, newValue) -> {
```
Comment (jorenham) [inline lifetiles-tree/src/main/java/nl/tudelft/lifetiles/tree/controller/TreeController.java]: Redundant curly brackets

File: lifetiles-graph/src/main/java/nl/tudelft/lifetiles/graph/controller/GraphController.java (hunk 6 above)
Comment (rutgervandenberg) [inline lifetiles-graph/src/main/java/nl/tudelft/lifetiles/graph/controller/GraphController.java:239]: Formatter does this. If I manually change it then the next time someone modifies this file and runs the formatter(as they should), it'll be reverted again.

File: lifetiles-sequence/src/main/java/nl/tudelft/lifetiles/sequence/controller/SequenceController.java (hunk 9 above)
Comment (rutgervandenberg) [inline lifetiles-sequence/src/main/java/nl/tudelft/lifetiles/sequence/controller/SequenceController.java:115]: Formatter does this. If I manually change it then the next time someone modifies this file and runs the formatter(as they should), it'll be reverted again.

**Comment by alangerak** [comment]
Looks good

**Comment by jorenham** [comment]
:+1:

**Comment by joswinter** [comment]
looks good

**Pull request merged**